packages/cli/src/lib/plugins/fingerprint.ts

```diff
@@ -17,28 +20,172 @@ type NativeFingerprintCommandOptions = {
+      }
+    }
+  }
+}
+
+function parseEnvFile(content: string): Record<string, string> {
+  const envVars: Record<string, string> = {};
+  const lines = content.split('\n');
+  
+  for (const line of lines) {
+    const trimmedLine = line.trim();
+    
+    if (trimmedLine && !trimmedLine.startsWith('#')) {
+      const [key, ...valueParts] = trimmedLine.split('=');
+      if (key && valueParts.length > 0) {
+        const value = valueParts.join('=').trim();
+        const cleanValue = value.replace(/^["']|["']$/g, '');
+        envVars[key.trim()] = cleanValue;
+      }
+    }
+  }
+  
+  return envVars;
+}
+
+function getEnvVariableHash(envVar: string): string | null {
+  const value = process.env[envVar];
+  if (value === undefined) {
+    return null;
+  }
+  return createHash('sha256').update(`${envVar}=${value}`).digest('hex');
+}
+
+function getEnvSources(envVars: string[]): EnvFingerprintSource[] {
+  return envVars.map((envVar) => ({
+    type: 'env' as const,
+    name: envVar,
+    value: process.env[envVar],
+    hash: getEnvVariableHash(envVar),
+  }));
+}
+
+function combineFingerprints(
+  nativeHash: string,
+  envSources: EnvFingerprintSource[],
+): string {
+  const validEnvHashes = envSources
+    .filter((source) => source.hash !== null)
+    .map((source) => source.hash);
+  
+  if (validEnvHashes.length === 0) {
+    return nativeHash;
+  }
+  
+  const combinedInput = [nativeHash, ...validEnvHashes].join(':');
+  return createHash('sha256').update(combinedInput).digest('hex');
+}
+
+function getEnvVarsForPlatform(platform: 'ios' | 'android'): string[] {
+  const platformVars = DEFAULT_ENV_VARS[platform] || [];
+  const commonVars = DEFAULT_ENV_VARS.common || [];
+  return [...commonVars, ...platformVars];
+}
+
+function extractEnvFromConfig(
+  fingerprintOptions: FingerprintSources & { env?: EnvConfig },
+  platform: 'ios' | 'android'
+): string[] {
+  if (!fingerprintOptions?.env) {
+    return getEnvVarsForPlatform(platform);
+  }
+  
+  const commonVars = fingerprintOptions.env.common || [];
+  const platformVars = fingerprintOptions.env[platform] || [];
+  
+  const uniqueVars = new Set([...commonVars, ...platformVars]);
+  return Array.from(uniqueVars);
+}
+
 export async function nativeFingerprintCommand(
   path: string,
-  { extraSources, ignorePaths }: FingerprintSources,
+  fingerprintOptions: FingerprintSources & { env?: EnvConfig },
   options: NativeFingerprintCommandOptions,
 ) {
   validateOptions(options);
   const platform = options.platform;
   const readablePlatformName = platform === 'ios' ? 'iOS' : 'Android';
+  
+  loadEnvFiles(path);
+  
+  const envVars = extractEnvFromConfig(fingerprintOptions, platform);
+  const envSources = getEnvSources(envVars);
 
   if (options.raw || !isInteractive()) {
     const fingerprint = await nativeFingerprint(path, {
       platform,
-      extraSources,
-      ignorePaths,
+      extraSources: fingerprintOptions.extraSources,
```

Comment:
instead of combining fingerprints later, we can put `envSources` into `extraSources` and still rely on the output of `nativeFingerprint`